tests/Unit/inc/classes/subscriber/Optimization/BufferSubscriber/redirectCanonical.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace WP_Rocket\Tests\Unit\inc\classes\subscriber\Optimization\BufferSubscriber;
+
+use WP_Rocket\Buffer\Optimization;
+use WP_Rocket\Subscriber\Optimization\Buffer_Subscriber;
+use WP_Rocket\Tests\Unit\TestCase;
+use Brain\Monkey\Functions;
+
+/**
+ * @covers \WP_Rocket\Subscriber\Optimization\Buffer_Subscriber::redirect_canonical
+ * @group Subscriber
+ */
+class Test_RedirectCanonical extends TestCase
+{
+	protected $subscriber;
+	protected $optimizer;
+
+	public function setUp(): void
+	{
+		$this->optimizer = \Mockery::mock(Optimization::class);
+		$this->subscriber = new Buffer_Subscriber($this->optimizer);
+		parent::setUp();
+	}
+
+	public function testShouldCallRedirect() {
+		Functions\expect('redirect_canonical')->with();
```

Comment:
```suggestion
		Functions\expect('redirect_canonical')->once();
```

Reply:
@Tabrisrp I don't get this one with no parameters is not more precise that once?